src/lcd.c

```diff
@@ -137,6 +139,33 @@ void myGPIOB_Init(void){
     GPIO_Init(GPIOB, &GPIO_InitStruct);
 }
 
+void LCD_CustomCharInit(void){
+  char customChar[8] = {
+	  0b10001,
+    0b11111,
+	  0b10001,
+	  0b01110,
+	  0b00010,
+	  0b00100,
+	  0b01000,
+	  0b01110
+  };
+  // resets the RS and R/W pins
+  LCD_SendWord(LCD_COMMAND, 0x00);
+  // Sets CG RAM address
+  LCD_SendWord(LCD_COMMAND, 0x40);
+  // all subsequent data is written and auto incremented to CG RAM address
+  int a;
```

Comment:
Initialize this variable in your `for` loop so its scope is confined to the loop.